src/plugins/dashboard/public/application/actions/filters_notification_badge.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+
+import { CoreStart, OverlayStart } from '@kbn/core/public';
+import {
+  EditPanelAction,
+  EmbeddableStart,
+  isFilterableEmbeddable,
+} from '@kbn/embeddable-plugin/public';
+import { createKibanaReactContext } from '@kbn/kibana-react-plugin/public';
+
+import { type AggregateQuery } from '@kbn/es-query';
+import { Action, IncompatibleActionError } from '../../services/ui_actions';
+import { toMountPoint } from '../../services/kibana_react';
+import { IEmbeddable, isErrorEmbeddable } from '../../services/embeddable';
+import { dashboardFilterNotificationBadge } from '../../dashboard_strings';
+
+export const BADGE_FILTERS_NOTIFICATION = 'ACTION_FILTERS_NOTIFICATION';
+
+export interface FiltersNotificationActionContext {
+  embeddable: IEmbeddable;
+}
+
+export class FiltersNotificationBadge implements Action<FiltersNotificationActionContext> {
+  public readonly id = BADGE_FILTERS_NOTIFICATION;
+  public readonly type = BADGE_FILTERS_NOTIFICATION;
+  public readonly order = 2;
+
+  private displayName = dashboardFilterNotificationBadge.getDisplayName();
+  private icon = 'filter';
+
+  constructor(
+    private application: CoreStart['application'],
+    private embeddable: EmbeddableStart,
```

Comment:
I actually originally had that but switched - wasn't really sure what was best style-wise, haha :) I'll go with renaming to `embeddableService` so that `contextEmbeddable` can just be `embeddable` - that makes the most sense to me I think 👍